test/regression/riscv/RISCVRegressionTest.hh

```diff
@@ -53,49 +53,102 @@ inline std::string paramToString(
   return coreString;
 }
 
-/** A helper macro to run a snippet of RISCV assembly code, returning from
+/** A helper macro to run a snippet of RISC-V assembly code, returning from
  * the calling function if a fatal error occurs. Four bytes containing zeros
  * are appended to the source to ensure that the program will terminate with
  * an unallocated instruction encoding exception instead of running into the
  * heap. */
-#define RUN_RISCV(source)                             \
-  {                                                   \
-    std::string sourceWithTerminator = source;        \
-    sourceWithTerminator += "\n.word 0";              \
-    run(sourceWithTerminator.c_str(), "+m,+a,+f,+d"); \
-  }                                                   \
+#define RUN_RISCV(source)                      \
+  {                                            \
+    std::string sourceWithTerminator = source; \
+    sourceWithTerminator += "\n.word 0";       \
+    run(sourceWithTerminator.c_str(), false);  \
+  }                                            \
   if (HasFatalFailure()) return
 
-/** A helper macro to run a snippet of RISCV assembly code, returning from
+/** A helper macro to run a snippet of RISC-V assembly code, returning from
  * the calling function if a fatal error occurs. Four bytes containing zeros
  * are appended to the source to ensure that the program will terminate with
  * an illegal instruction exception instead of running into the heap. This
  * specifically targets the compressed extension allowing for the RUN_RISCV
  * macro to ignore it, otherwise LLVM eagerly emits compressed instructions for
  * non-compressed assembly. */
-#define RUN_RISCV_COMP(source)                           \
-  {                                                      \
-    std::string sourceWithTerminator = source;           \
-    sourceWithTerminator += "\n.word 0";                 \
-    run(sourceWithTerminator.c_str(), "+m,+a,+f,+d,+c"); \
-  }                                                      \
+#define RUN_RISCV_COMP(source)                 \
+  {                                            \
+    std::string sourceWithTerminator = source; \
+    sourceWithTerminator += "\n.word 0";       \
+    run(sourceWithTerminator.c_str(), true);   \
+  }                                            \
   if (HasFatalFailure()) return
 
-/** The test fixture for all RISCV regression tests. */
+/** A helper macro to predecode the first instruction in a snippet of RISC-V
+ * assembly code and check the assigned group(s) for each micro-op matches the
+ * expected group(s). Returns from the calling function if a fatal error occurs.
+ * Four bytes containing zeros are appended to the source to ensure that the
+ * program will terminate with an unallocated instruction encoding exception
+ * instead of running into the heap.
+ */
+#define EXPECT_GROUP(source, ...)                                   \
```

Comment:
Is this the same as the macro defined in the aarch64RegressionTest.hh file?  If so, could it be moved to the generic RegressionTest.hh file to avoid duplication?

Reply:
These are not the same. They call their respective checkGroup function which is ISA specific. Both initialise LLVM for their respective ISA and the RISC-V version takes an extra parameter for compressed instructions